pandas/tests/io/test_html.py

```diff
@@ -448,6 +438,44 @@ def test_header_and_one_column(self):
         result = self.read_html(data)[0]
         tm.assert_frame_equal(result, expected)
 
+    def test_thead_without_tr(self):
+        """
+        Ensure parser adds <tr> within <thead> on malformed HTML.
+        """
+        data1 = StringIO('''<table>
+            <thead>
+                <tr>
+                    <th>Country</th>
+                    <th>Municipality</th>
+                    <th>Year</th>
+                </tr>
+            </thead>
+            <tbody>
+                <tr>
+                    <td>Ukraine</td>
+                    <th>Odessa</th>
+                    <td>1944</td>
+                </tr>
+            </tbody>
+        </table>''')
+        data2 = StringIO('''<table>
```

Comment:
Generally can add spacing between variables to make the tests more readable

Reply:
done in lots and lots of tests